base/reduce.jl

```diff
@@ -318,6 +318,9 @@ reduce_empty(::typeof(*), ::Type{<:AbstractChar}) = ""
 reduce_empty(::typeof(&), ::Type{Bool}) = true
 reduce_empty(::typeof(|), ::Type{Bool}) = false
 
+reduce_empty(::typeof(max), T::Type{<:AbstractFloat}) = typemin(T)
+reduce_empty(::typeof(min), T::Type{<:AbstractFloat}) = typemax(T)
```

Comment:
Why not just let `typemin`/`typemax` throw an error `MethodError: no method matching typemax(::Type{T})`?

```suggestion
reduce_empty(::typeof(max), T) = typemin(T)
reduce_empty(::typeof(min), T) = typemax(T)
```